storages/backends/azure_storage.py

```diff
@@ -24,6 +24,20 @@
 from storages.utils import setting
 from storages.utils import to_bytes
 
+try:
+    from django_guid import get_guid
+except ImportError:
+    def get_guid():
+        return None
+
+
+def _make_headers():
+    headers = {}
+    correlation_id = get_guid()
```

Comment:
Is committing to django-guid for this something we want? What if people have other means of generating a Correlation ID (django-log-request-id or OpenTelemetry).

Maybe simple adding a `make_headers` method to the Azure Storage, and allowing users to override it using any solution they want, would be a simpler solution, without committing to any individual package.